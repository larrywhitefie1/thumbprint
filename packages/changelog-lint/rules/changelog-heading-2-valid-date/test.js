const test = require('../../test-utils');
const rule = require('./');

const valid = ['## 2012-03-03'];

const invalid = [
    '## 2012-03-90',
    '## 1.1.1 - 20133-01-01',
    '## 1.1.1 - 2013-13-01',
    '## 1.1.1 - 2013-12-32',
    '## 1.1.1 - 2013-1-20',
    '## 1.1.1 - 2013-01-1',
    '## 1.1.1 - 2013-1-1',
];

test({
    rule,
    valid,
    invalid,
});
