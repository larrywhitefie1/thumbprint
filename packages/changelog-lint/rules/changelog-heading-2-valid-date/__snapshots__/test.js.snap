// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`invalid case #0 fails 1`] = `
Object {
  "input": "## 2012-03-90",
  "output": Array [
    Object {
      "column": 1,
      "fatal": false,
      "line": 1,
      "location": Object {
        "end": Object {
          "column": 14,
          "line": 1,
          "offset": 13,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "reason": "“2012-03-90“ does not end with a valid date in the format YYYY-MM-DD.",
      "ruleId": "changelog-heading-2-valid-date",
      "source": "remark-lint",
      "stack": null,
    },
  ],
}
`;

exports[`invalid case #1 fails 1`] = `
Object {
  "input": "## 1.1.1 - 20133-01-01",
  "output": Array [
    Object {
      "column": 1,
      "fatal": false,
      "line": 1,
      "location": Object {
        "end": Object {
          "column": 23,
          "line": 1,
          "offset": 22,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "reason": "“1.1.1 - 20133-01-01“ does not end with a valid date in the format YYYY-MM-DD.",
      "ruleId": "changelog-heading-2-valid-date",
      "source": "remark-lint",
      "stack": null,
    },
  ],
}
`;

exports[`invalid case #2 fails 1`] = `
Object {
  "input": "## 1.1.1 - 2013-13-01",
  "output": Array [
    Object {
      "column": 1,
      "fatal": false,
      "line": 1,
      "location": Object {
        "end": Object {
          "column": 22,
          "line": 1,
          "offset": 21,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "reason": "“1.1.1 - 2013-13-01“ does not end with a valid date in the format YYYY-MM-DD.",
      "ruleId": "changelog-heading-2-valid-date",
      "source": "remark-lint",
      "stack": null,
    },
  ],
}
`;

exports[`invalid case #3 fails 1`] = `
Object {
  "input": "## 1.1.1 - 2013-12-32",
  "output": Array [
    Object {
      "column": 1,
      "fatal": false,
      "line": 1,
      "location": Object {
        "end": Object {
          "column": 22,
          "line": 1,
          "offset": 21,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "reason": "“1.1.1 - 2013-12-32“ does not end with a valid date in the format YYYY-MM-DD.",
      "ruleId": "changelog-heading-2-valid-date",
      "source": "remark-lint",
      "stack": null,
    },
  ],
}
`;

exports[`invalid case #4 fails 1`] = `
Object {
  "input": "## 1.1.1 - 2013-1-20",
  "output": Array [
    Object {
      "column": 1,
      "fatal": false,
      "line": 1,
      "location": Object {
        "end": Object {
          "column": 21,
          "line": 1,
          "offset": 20,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "reason": "“1.1.1 - 2013-1-20“ does not end with a valid date in the format YYYY-MM-DD.",
      "ruleId": "changelog-heading-2-valid-date",
      "source": "remark-lint",
      "stack": null,
    },
  ],
}
`;

exports[`invalid case #5 fails 1`] = `
Object {
  "input": "## 1.1.1 - 2013-01-1",
  "output": Array [
    Object {
      "column": 1,
      "fatal": false,
      "line": 1,
      "location": Object {
        "end": Object {
          "column": 21,
          "line": 1,
          "offset": 20,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "reason": "“1.1.1 - 2013-01-1“ does not end with a valid date in the format YYYY-MM-DD.",
      "ruleId": "changelog-heading-2-valid-date",
      "source": "remark-lint",
      "stack": null,
    },
  ],
}
`;

exports[`invalid case #6 fails 1`] = `
Object {
  "input": "## 1.1.1 - 2013-1-1",
  "output": Array [
    Object {
      "column": 1,
      "fatal": false,
      "line": 1,
      "location": Object {
        "end": Object {
          "column": 20,
          "line": 1,
          "offset": 19,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "reason": "“1.1.1 - 2013-1-1“ does not end with a valid date in the format YYYY-MM-DD.",
      "ruleId": "changelog-heading-2-valid-date",
      "source": "remark-lint",
      "stack": null,
    },
  ],
}
`;
